backend/src/test/java/heartbeat/component/E2EUrlGeneratorTest.java

```diff
@@ -0,0 +1,20 @@
+package heartbeat.component;
+
+import org.junit.jupiter.api.Test;
+
+import java.net.URI;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class E2EUrlGeneratorTest {
+
+	private final E2EUrlGenerator e2EUrlGenerator = new E2EUrlGenerator();
+
+	@Test
+	public void testGetUri() {
+		e2EUrlGenerator.setUrl("mockTest");
```

Comment:
Could use ReflectionTestUtils to set the url in test, do not need add "@setter" in E2EUrlGenerator.
```
class E2EUrlGeneratorTest {

	@Test
	public void testGetUri() {
		String url = "http://test.com";
		E2EUrlGenerator e2EUrlGenerator = new E2EUrlGenerator();
		ReflectionTestUtils.setField(e2EUrlGenerator, "url", url);

		URI uri = e2EUrlGenerator.getUri("test-site");

		assertEquals(url, uri.toString());
	}

}
```